src/test/java/net/sf/jabref/gui/keyboard/KeyBindingsDialogViewModelTest.java

```diff
@@ -0,0 +1,175 @@
+        assertFalse(keyBindingsPreferences.checkKeyCombinationEquality(KeyBinding.COPY, shortcutKeyEvent));
+        model.grabKey(shortcutKeyEvent);
+        KeyCombination combination = KeyCombination.keyCombination(keyBindingRepository.get(KeyBinding.COPY).get());
+        assertFalse(keyBindingsPreferences.checkKeyCombinationEquality(combination, shortcutKeyEvent));
+        model.saveKeyBindings();
+        assertFalse(keyBindingsPreferences.checkKeyCombinationEquality(KeyBinding.COPY, shortcutKeyEvent));
+    }
+
+    @Test
+    public void testNoModifierValidKeyBinding() {
+        KeyEvent shortcutKeyEvent = prepareNewKeyBind(KeyBinding.IMPORT_INTO_NEW_DATABASE, "F1", KeyCode.F1, false,
+                false, false, false);
+        assertFalse(keyBindingsPreferences.checkKeyCombinationEquality(KeyBinding.IMPORT_INTO_NEW_DATABASE,
+                shortcutKeyEvent));
+        model.grabKey(shortcutKeyEvent);
+        KeyCombination combination = KeyCombination
+                .keyCombination(keyBindingRepository.get(KeyBinding.IMPORT_INTO_NEW_DATABASE).get());
+        assertTrue(keyBindingsPreferences.checkKeyCombinationEquality(combination, shortcutKeyEvent));
+        model.saveKeyBindings();
+        assertTrue(keyBindingsPreferences.checkKeyCombinationEquality(KeyBinding.IMPORT_INTO_NEW_DATABASE,
+                shortcutKeyEvent));
+    }
+
+    @Test
+    public void testKeyBindingCategory() {
+        KeyBindingViewModel bindViewModel = new KeyBindingViewModel(KeyBindingCategory.FILE);
+        model.getSelectedKeyBinding().set(new TreeItem<>(bindViewModel));
+        KeyEvent shortcutKeyEvent = new KeyEvent(KeyEvent.KEY_PRESSED, "M", "M", KeyCode.M, true, true, true, false);
+        assertFalse(keyBindingsPreferences.checkKeyCombinationEquality(KeyBinding.CLEANUP, shortcutKeyEvent));
+        model.grabKey(shortcutKeyEvent);
+        assertNull(model.getSelectedKeyBinding().get().getValue().getKeyBinding());
+    }
+
+    @Test
+    public void testNewKeyKeyBindingInRepository() {
+        KeyEvent shortcutKeyEvent = prepareNewKeyBind(KeyBinding.CLEANUP, "K", KeyCode.K, true, true, true, false);
+        assertFalse(keyBindingsPreferences.checkKeyCombinationEquality(KeyBinding.CLEANUP, shortcutKeyEvent));
+        model.grabKey(shortcutKeyEvent);
+
+        KeyEvent shortcutKeyEvent1 = prepareNewKeyBind(KeyBinding.FORWARD, "L", KeyCode.L, true, false, false, false);
+        assertFalse(keyBindingsPreferences.checkKeyCombinationEquality(KeyBinding.FORWARD, shortcutKeyEvent1));
+        model.grabKey(shortcutKeyEvent1);
+
+        KeyEvent shortcutKeyEvent2 = prepareNewKeyBind(KeyBinding.OPEN_FOLDER, "R", KeyCode.R, true, false, true,
+                false);
+        assertFalse(keyBindingsPreferences.checkKeyCombinationEquality(KeyBinding.OPEN_FOLDER, shortcutKeyEvent2));
+        model.grabKey(shortcutKeyEvent2);
```

Comment:
Split into 3 tests. In general, test only one thing in a test (so at best you have only one assert per test)